java/src/jni/h5util.c

```diff
@@ -873,7 +873,7 @@ h5str_sprintf(JNIEnv *env, h5str_t *out_str, hid_t container, hid_t tid, void *i
                     if (NULL == (this_str = (char *)HDmalloc(typeSize + 1)))
                         H5_OUT_OF_MEMORY_ERROR(ENVONLY, "h5str_sprintf: failed to allocate string buffer");
 
-                    HDstrncpy(this_str, tmp_str, typeSize);
+                    HDstrcpy(this_str, tmp_str);
```

Comment:
We already know the size of the buffer; don't use a strncpy with length equal to the buffer size.

Reply:
Could just use a strdup in this case.

Reply:
Sure, I was just going for the minimal amount of changes necessary here.

Reply:
Fixed